spec/v2/providers/storage.spec.ts

```diff
@@ -0,0 +1,336 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import * as config from '../../../src/config';
+import * as options from '../../../src/v2/options';
+import * as storage from '../../../src/v2/providers/storage';
+import { FULL_OPTIONS, FULL_TRIGGER } from './helpers';
+
+describe('v2/storage', () => {
+  describe('_getOptsAndBucket', () => {
+    it('should return throw error without a bucket', () => {
+      expect(() => storage._getOptsAndBucket()).to.throw();
+    });
+
+    it('should return the default bucket on no params', () => {
+      const configStub = sinon
+        .stub(config, 'firebaseConfig')
+        .returns({ storageBucket: 'default-bucket' });
+
+      const [opts, bucket] = storage._getOptsAndBucket();
+
+      configStub.restore();
+      expect(opts).to.deep.equal({});
+      expect(bucket).to.eq('default-bucket');
+    });
+
+    it('should return the given bucket', () => {
+      const [opts, bucket] = storage._getOptsAndBucket('my-bucket');
+
+      expect(opts).to.deep.equal({});
+      expect(bucket).to.eq('my-bucket');
+    });
+
+    it('should return the given bucket and opts', () => {
+      const [opts, bucket] = storage._getOptsAndBucket({
+        bucket: 'my-bucket',
+        region: 'us-west1',
+      });
+
+      expect(opts).to.deep.equal({ region: 'us-west1' });
+      expect(bucket).to.eq('my-bucket');
+    });
+  });
+
+  describe('_onOperation', () => {
+    const EVENT_TRIGGER = {
+      eventType: 'google.cloud.storage.event-type',
+      resource: 'my-bucket',
+      service: 'storage.googleapis.com',
+    };
+
+    beforeEach(() => {
+      options.setGlobalOptions({});
+      process.env.GCLOUD_PROJECT = 'aProject';
```

Comment:
Hmm can you remind me why setting this environment variable is required for the test to work?

Reply:
If we don't set it, then when we try and perform `optionsToTriggerAnnotations` it will call `serviceAccountFromShorthand` which requires this env var to be set and throw an error otherwise